nemo/utils/app_state.py

```diff
@@ -0,0 +1,133 @@
+# -*- coding: utf-8 -*-
+# =============================================================================
+# Copyright (c) 2020 NVIDIA. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# =============================================================================
+
+import threading
+
+# Sadly have to import this to avoid circular dependencies.
+import nemo
+
+
+class Singleton(type):
```

Comment:
Does this replace https://github.com/NVIDIA/NeMo/blob/master/nemo/utils/metaclasses.py?

Reply:
Yeah, thanks for reminding to unify those two, will do it ASAP